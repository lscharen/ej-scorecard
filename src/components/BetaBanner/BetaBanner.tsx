import { FormattedMessage, FormattedDate } from 'gatsby-plugin-intl';
import * as styles from './BetaBanner.module.scss';

const METH_1_0_RELEASE_DATE = new Date(2022, 10, 22, 11, 59, 59); // Nov 22 2022
export const boldFn = (chunks: string) => <strong>{chunks}</strong>;

const BETA_BANNER_CONTENT = (
  <FormattedMessage
    id={`common.pages.alerts.banner.beta.content`}
    defaultMessage={`<bold1>This tool has been updated.</bold1> The 1.0 version of the tool was released on {relDate}.`}
    description={`Alert body that appears on landing page.`}
    values={{
      bold1: boldFn,
      relDate: (
        <FormattedDate
          value={METH_1_0_RELEASE_DATE}
          year="numeric"
          month="short"
          day="numeric"
        />
      ),
    }}
  />
);

const BetaBanner = () => {
  return (
    <div className={styles.betaBannerContainer}>
      <div className={styles.betaBanner}>{BETA_BANNER_CONTENT}</div>
    </div>
  );
};

export default BetaBanner;
