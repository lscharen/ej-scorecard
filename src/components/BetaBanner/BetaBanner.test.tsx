import { render } from '@testing-library/react';
import BetaBanner from './BetaBanner';

describe(`rendering of the BetaBanner`, () => {
  const { asFragment } = render(<BetaBanner />);

  it(`checks if component renders`, () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
