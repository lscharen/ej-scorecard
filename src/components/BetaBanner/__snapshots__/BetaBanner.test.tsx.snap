// Vitest Snapshot v1

exports[`rendering of the BetaBanner > checks if component renders 1`] = `
<DocumentFragment>
  <div>
    <div>
      This is a new tool.
    </div>
  </div>
</DocumentFragment>
`;
